providers/azure/helpers.go

```diff
@@ -64,11 +66,12 @@ func getContainerClient(conf Config) (*container.Client, error) {
 	}
 
 	// Use MSI for authentication.
-	msiOpt := &azidentity.ManagedIdentityCredentialOptions{}
 	if conf.UserAssignedID != "" {
-		msiOpt.ID = azidentity.ClientID(conf.UserAssignedID)
+		if err := os.Setenv("AZURE_CLIENT_ID", conf.UserAssignedID); err != nil {
+			return nil, errors.Wrapf(err, "unable to set environment variable for AZURE_CLIENT_ID")
+		}
 	}
-	cred, err := azidentity.NewManagedIdentityCredential(msiOpt)
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
Could we keep it and just use a ChainedTokenCredential with ManagedIdentityCredential and WorkloadIdentityCredential inside?

Reply:
That way we would not need the environment variable and it feels a bit cleaner

Reply:
The aws config has a `aws_sdk_auth` which is pretty similar to `DefaultAzureCredential`. Should we maybe keep the managed identity as is but add another config option `azure_default_credential`? Or maybe just have that be the fallback if no other methods are valid?

Reply:
Yeah actually ended up going with that approach pretty much haha. `ChainedTokenCredential` didn't end up working properly, so have just stuck with using `ManagedIdentityCredential` if `UserAssignedID` assigned is set and use `WorkloadIdentityCredential` otherwise. Have tested both types of authentication and it's working well.

Reply:
Actually, your approach makes more sense now that I think about it. Rather than falling back to `WorkloadIdentityCredential` we can fall back to `DefaultAzureCredential`. Will test it now.

Reply:
Cool. That way we have access to all auth mechanisms.

Reply:
Just finished testing and it's working well. Have pushed the changes.